qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/database/DatabaseSession.groovy

```diff
@@ -24,61 +26,37 @@ class DatabaseSession {
 
     protected static DatabaseSession INSTANCE
 
-    private final String hostname
-    private final String url
-    private final String port
-    private final String sql_database
-    private final String username
-    private final String password
+    private BasicDataSource dataSource
 
     private static final Logger LOG = LogManager.getLogger(DatabaseSession.class)
 
     //create session with DatabaseSession.init() and then DatabaseSession.INSTANCE to use the created session
     private DatabaseSession(String user, String password, String host, String port, String sql_database) {
-        username = user
-        this.password = password
-        this.hostname = host
-        this.port = port
-        this.sql_database = sql_database
-        this.url = "jdbc:mysql://" + host + ":" + port + "/" + sql_database
-        String mysqlDriverName = "com.mysql.jdbc.Driver"
-        try {
-            Class.forName(mysqlDriverName)
-        } catch (ClassNotFoundException e) {
-            e.printStackTrace()
-        }
-        LOG.info("MySQL Database instance created")
-    }
 
-    /**
-     *
-     * @return
-     * @throws SQLException
-     */
-    static Connection getDatabaseInstanceAlternative(Properties config) throws SQLException {
-        try{
-            Class.forName("com.mysql.jdbc.Driver")
-            return DriverManager.getConnection("jdbc:mysql://" + config.getProperty("mysql.host") + ":" + config.getProperty("mysql.port") + "/" + config.getProperty("mysql.db"),
-                    config.getProperty("mysql.user"), config.getProperty("mysql.pass"))
-        }
-        catch (Exception e){
-            LOG.error "Could not establish JBDC connection"
-            e.printStackTrace()
-        }
-        return null
+        def url = "jdbc:mysql://" + host + ":" + port + "/" + sql_database
+
+        dataSource = new BasicDataSource()
+        dataSource.setUrl(url)
+        dataSource.setUsername(user)
+        dataSource.setPassword(password)
+        dataSource.setMinIdle(5)
+        dataSource.setMaxIdle(10)
+        dataSource.setMaxOpenPreparedStatements(100)
+
+        LOG.info("MySQL Database instance created")
     }
 
 
     /**
      * Initiates the database connection by retrieving configuration information from the {@link ConfigurationManager}
      * The instance is only created if there is no other existing
      */
-    static void init() {
+    static void create() {
         ConfigurationManager conf = ConfigurationManagerFactory.getInstance()
```

Comment:
how do we handle local sessions? Afaik the ConfigurationManager only works with a Liferayinstance

Reply:
no, this is handled by the Configuration Manager:

https://github.com/qbicsoftware/portal-utils-lib/blob/development/src/main/java/life/qbic/portal/utils/ConfigurationManagerFactory.java

Reply:
In case if the running code is not deployed as a portlet in a Liferay env, the factory checks for developer property file and creates a config from this

Reply:
Then I guess there was another problem when I tried to set it up